spec/services/virtual_merge_tool_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe VirtualMergeTool do
+  let(:parent) do
+    Dor::Item.create do |item|
+      item.contentMetadata.content = <<~XML
+        <contentMetadata></contentMetadata>
+      XML
+    end
+  end
+
+  let(:child1) do
+    Dor::Item.create do |item|
+      item.contentMetadata.content = <<~XML
+        <contentMetadata>
+          <resource id="bb000kg4251_1" sequence="1" type="image">
+            <file id="bb000kg4251.jpg" mimetype="image/jpeg" size="1347965" preserve="yes" publish="no" shelve="no">
+            </file>
+          </resource>
+        </contentMetadata>
+      XML
+    end
+  end
+
+  let(:child2) do
+    Dor::Item.create do |item|
+      item.contentMetadata.content = <<~XML
+        <contentMetadata>
+          <resource id="bb000ff1111_1" sequence="1" type="image">
+            <file id="bb000ff1111.jpg" mimetype="image/jpeg" size="999" preserve="yes" publish="no" shelve="no">
+            </file>
+          </resource>
+        </contentMetadata>
+      XML
+    end
+  end
+
+  describe '#run' do
+    subject(:run) { instance.run }
+
+    let(:instance) do
+      described_class.new(parent_druid: parent.id, child_druids: [child1.id, child2.id])
+    end
+    let(:namespaceless) { parent.id.sub('druid:', '') }
+
+    context 'when the parent is open for modification' do
+      before do
+        run
+      end
+
+      it 'merges objects' do
+        parent.reload
+        expect(parent.contentMetadata.content).to be_equivalent_to <<~XML
+          <contentMetadata>
+            <resource sequence="1" id="#{namespaceless}_1" type="image">
+              <relationship type="alsoAvailableAs" objectId="#{child1.id}"/>
+            </resource>
+            <resource sequence="2" id="#{namespaceless}_2" type="image">
+              <relationship type="alsoAvailableAs" objectId="#{child2.id}"/>
+            </resource>
+          </contentMetadata>
+        XML
+        child1.reload
+        expect(child1.object_relations[:is_constituent_of]).to eq ["info:fedora/#{parent.id}"]
+        child2.reload
+        expect(child2.object_relations[:is_constituent_of]).to eq ["info:fedora/#{parent.id}"]
+      end
```

Comment:
I think we might want tests for:
- parent isn't open for modification
- parent ok, but one child isn't ... show other child added?
- useful error information going into logs for diff errors (to ultimately be viewed by user)